02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/02 Tensorflow/07 Save Models.html

```diff
@@ -0,0 +1,27 @@
+<p>Follow these steps to save <code>Tensorflow</code> models into the <a href="https://www.quantconnect.com/docs/v2/writing-algorithms/object-store">ObjectStore</a>:</p>
+
+<ol>
+    <li>Export the <code>TensorFlow</code> graph as a JSON object.</li>
+    <div class="section-example-container">
+        <pre class="python">graph_definition = tf.compat.v1.train.export_meta_graph()
+json_graph = json_format.MessageToJson(graph_definition)</pre>
+    </div>
+
+    <li>Export the <code>TensorFlow</code> weights as a JSON object.</li>
+    <div class="section-example-container">
+        <pre class="python"># Define a function to get the weights from the tensorflow session
```

Comment:
I think we should remove this internal function definition. If we just use the body of the definition, we can reduce the logic down to just 3 lines of code.

```
    weights = self.model.run(tf.compat.v1.trainable_variables())
    weights = [w.tolist() for w in weights]
    json_weights = json5.dumps(weights)
```